ui/src/views/login/Login.jsx

```diff
@@ -27,20 +27,24 @@ export default function Login() {
   }, []);
 
   const tryLogin = async () => {
-    if (username.length === 0 || password.length === 0) {
-      setError('Username and password are mandatory.');
+    if (!username?.trim() || !password) {
```

Comment:
Tbh, I thought about using a toast here as well, but found that showing the in the login form directly looks nicer, therefor I think using a toast here is not ideal